### ISSUE TITLE
Retaining checkbox values upon validation failure

### ISSUE THREAD

**Issue by susanBuck** [susanBuck/dwa15-fall2017#99]
We saw in Week 8's lectures that it's easy to retain our text input values when validation fails using the `old` helper method:

```php
<input type='text' name='title' id='title' value='{{ old('title') }}'>
```

When it comes to other more complex input types, though, it requires some more work. 

For example, how can you use `old` to retain the values for a group of checkboxes?

Here's an example how:
```php
<input type='checkbox' name='example[]' value='a' {{ (old('example') and in_array('a', old('example'))) ? 'checked' : ''}} >
<input type='checkbox' name='example[]' value='b' {{ (old('example') and in_array('b', old('example'))) ? 'checked' : ''}} >
<input type='checkbox' name='example[]' value='c' {{ (old('example') and in_array('c', old('example'))) ? 'checked' : ''}} >
```

The key part is these lines:
```php
{{ (old('example') and in_array('a', old('example'))) ? 'checked' : ''}}
```

In plain English, this says:

*if* `old('example')` exists *and* we see the presence of `a` in this data, then echo `checked`. Otherwise, echo nothing.

**Comment by helhaggaz** [comment]
To retain a check box value upon validation failure I used 
`{{(old('example', $example)) ? 'CHECKED' : ''}}`
where `$example` is the value submitted on the form, and it worked.

**Comment by helhaggaz** [comment]
just noticed your post and it seems I used the recommended way by making use of the default value of `old` [https://github.com/susanBuck/dwa15-fall2017/issues/112](https://github.com/susanBuck/dwa15-fall2017/issues/112)
